crates/proof-of-sql/src/sql/proof_gadgets/monotonic.rs

```diff
@@ -0,0 +1,132 @@
+//! Prove that a column is increasing, strictly or non-strictly.
+use super::{
+    final_round_evaluate_shift, first_round_evaluate_shift, prover_evaluate_sign,
+    verifier_evaluate_sign, verify_shift,
+};
+use crate::{
+    base::{proof::ProofError, scalar::Scalar},
+    sql::proof::{FinalRoundBuilder, FirstRoundBuilder, VerificationBuilder},
+};
+use alloc::vec;
+use bumpalo::Bump;
+
+/// Perform first round evaluation of monotonicity.
+pub(crate) fn first_round_evaluate_monotonic<S: Scalar>(
+    builder: &mut FirstRoundBuilder<'_, S>,
+    num_rows: usize,
+) {
+    builder.produce_one_evaluation_length(num_rows + 1);
+    first_round_evaluate_shift(builder, num_rows);
+}
+
+/// Perform final round evaluation of monotonicity.
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn final_round_evaluate_monotonic<'a, S: Scalar>(
+    builder: &mut FinalRoundBuilder<'a, S>,
+    alloc: &'a Bump,
+    alpha: S,
+    beta: S,
+    column: &'a [S],
+    strictness: bool,
```

Comment:
Can we make this a const generic?

Reply:
@JayWhite2357 You mean make strictness a const parameter?

Reply:
Also, I think it would be pretty easy to add an `increasing` parameter as well so that we can make this any of
* strictly increasing
* non-strictly increasing
* strictly decreasing
* non-strictly decreasing